### ISSUE TITLE
Postgres startup fails

### ISSUE THREAD

**Issue by RX14** [docker-library/postgres#87]
```
db_1 | The files belonging to this database system will be owned by user "postgres".
db_1 | This user must also own the server process.
db_1 | 
db_1 | The database cluster will be initialized with locale "en_US.utf8".
db_1 | The default database encoding has accordingly been set to "UTF8".
db_1 | The default text search configuration will be set to "english".
db_1 | 
db_1 | Data page checksums are disabled.
db_1 | 
db_1 | fixing permissions on existing directory /var/lib/postgresql/data ... ok
db_1 | creating subdirectories ... ok
db_1 | selecting default max_connections ... 100
db_1 | selecting default shared_buffers ... 128MB
db_1 | selecting dynamic shared memory implementation ... posix
db_1 | creating configuration files ... ok
db_1 | creating template1 database in /var/lib/postgresql/data/base/1 ... ok
db_1 | initializing pg_authid ... ok
db_1 | initializing dependencies ... ok
db_1 | creating system views ... ok
db_1 | loading system objects' descriptions ... ok
db_1 | creating collations ... ok
db_1 | creating conversions ... ok
db_1 | creating dictionaries ... ok
db_1 | setting privileges on built-in objects ... ok
db_1 | creating information schema ... ok
db_1 | loading PL/pgSQL server-side language ... ok
db_1 | vacuuming database template1 ... ok
db_1 | copying template1 to template0 ... ok
db_1 | copying template1 to postgres ... ok
db_1 | syncing data to disk ... ok
db_1 | 
db_1 | WARNING: enabling "trust" authentication for local connections
db_1 | You can change this by editing pg_hba.conf or using the option -A, or
db_1 | --auth-local and --auth-host, the next time you run initdb.
db_1 | 
db_1 | Success. You can now start the database server using:
db_1 | 
db_1 |     postgres -D /var/lib/postgresql/data
db_1 | or
db_1 |     pg_ctl -D /var/lib/postgresql/data -l logfile start
db_1 | 
db_1 | ****************************************************
db_1 | WARNING: No password has been set for the database.
db_1 |          This will allow anyone with access to the
db_1 |          Postgres port to access your database. In
db_1 |          Docker's default configuration, this is
db_1 |          effectively any other container on the same
db_1 |          system.
db_1 | 
db_1 |          Use "-e POSTGRES_PASSWORD=password" to set
db_1 |          it in "docker run".
db_1 | ****************************************************
db_1 | waiting for server to start....FATAL:  data directory "/var/lib/postgresql/data" has wrong ownership
db_1 | HINT:  The server must be started by the user that owns the data directory.
db_1 | .... stopped waiting
db_1 | pg_ctl: could not start server
db_1 | Examine the log output.
```

```
$ docker -v
Docker version 1.8.1, build d12ea79
```

Started using docker-compose 0.4.0

**Comment by yosifkit** [comment]
Could you post your compose yaml file, `docker info`, and `docker version`?  Is this on a Mac using boot2docker?

**Comment by RX14** [comment]
```
$ docker info
Containers: 24
Images: 311
Storage Driver: devicemapper
 Pool Name: docker-8:17-522212-pool
 Pool Blocksize: 65.54 kB
 Backing Filesystem: <unknown>
 Data file: /dev/loop0
 Metadata file: /dev/loop1
 Data Space Used: 11.49 GB
 Data Space Total: 107.4 GB
 Data Space Available: 95.88 GB
 Metadata Space Used: 16.23 MB
 Metadata Space Total: 2.147 GB
 Metadata Space Available: 2.131 GB
 Udev Sync Supported: true
 Deferred Removal Enabled: false
 Data loop file: /data/docker/devicemapper/devicemapper/data
 Metadata loop file: /data/docker/devicemapper/devicemapper/metadata
 Library Version: 1.02.102 (2015-07-07)
Execution Driver: native-0.2
Logging Driver: json-file
Kernel Version: 4.1.6-1-ARCH
Operating System: Arch Linux
CPUs: 4
Total Memory: 7.638 GiB
Name: RX-MAIN1
ID: UPHF:6F33:E2AO:OSWY:H32N:XIVD:EJB4:5LRT:ZEBH:UFMP:VVY5:BQA7
```

```
$ docker version
Client:
 Version:      1.8.1
 API version:  1.20
 Go version:   go1.4.2
 Git commit:   d12ea79
 Built:        Sat Aug 15 17:29:10 UTC 2015
 OS/Arch:      linux/amd64

Server:
 Version:      1.8.1
 API version:  1.20
 Go version:   go1.4.2
 Git commit:   d12ea79
 Built:        Sat Aug 15 17:29:10 UTC 2015
 OS/Arch:      linux/amd64
```

```
$ cat docker-compose.yml 
db:
  image: postgres:9.4.4

api:
  build: .
```

This is on linux, and the container has no volumes (that docker-compose hasn't added)

**Comment by yosifkit** [comment]
You could try a `docker-compose rm -v` to make sure that volumes are cleaned up and then try starting it up again.  The postgres image does have a volume defined in the Dockerfile, so the volume will be created and docker-compose tries to preserve the data between runs.
